roles/openshift_health_checker/openshift_checks/logging/podlogs.py

```diff
@@ -0,0 +1,174 @@
+
+    name = "pod_log_query_time"
+    tags = ["health", "logging"]
+
+    logging_namespace = "logging"
+    pod_es_host = ES_DEFAULT_INDEX
+    pod_es_index = ES_DEFAULT_HOST
+
+    def run(self, tmp, task_vars):
+        """Add log entry by making unique request to Kibana. Check for unique entry in the ElasticSearch pod logs."""
+        self.pod_es_host = get_var(task_vars, "openshift_check_pod_elasticsearch_host", default=ES_DEFAULT_HOST)
+        self.pod_es_index = get_var(task_vars, "openshift_check_pod_elasticsearch_index", default=ES_DEFAULT_INDEX)
+
+        pod_log_timeout = int(get_var(task_vars, "openshift_check_pod_logs_timeout", default=ES_CMD_TIMEOUT_SECONDS))
+
+        # get all Kibana pods
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default=self.logging_namespace)
+        pods, error = super(PodLogQueryTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "kibana",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        running_kibana_pods = self.running_pods(pods)
+        if not len(running_kibana_pods):
+            msg = ('No Kibana pods were found to be in the "Running" state.'
+                   'At least one Kibana pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        # get all Elasticsearch pods
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default=self.logging_namespace)
+        pods, error = super(PodLogQueryTime, self).get_pods_for_component(
+            self.execute_module,
+            self.logging_namespace,
+            "es",
+            task_vars,
+        )
+
+        if error:
+            return {"failed": True, "changed": False, "msg": error}
+
+        running_es_pods = self.running_pods(pods)
+        if not len(running_es_pods):
+            msg = ('No Elasticsearch pods were found to be in the "Running" state.'
+                   'At least one Elasticsearch pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        error = self.check_log_delay(running_kibana_pods[0], running_es_pods[0], pod_log_timeout, task_vars)
+        if error:
+            return {"failed": True, "msg": error}
+
+        return {}
+
+    def check_log_delay(self, kibana_pod, es_pod, pod_log_timeout, task_vars):
+        """Check delay of logs and return any errors accordingly"""
+        error = None
+
+        try:
+            uid = self.curl_kibana_with_uuid(kibana_pod, task_vars)
+            self.wait_until_cmd_or_err(es_pod, uid, pod_log_timeout, task_vars)
+        except OpenShiftCheckException as error:
+            error = str(error)
+        except KeyError as error:
+            error = "Invalid json response from pod: {}".format(str(error))
```

Comment:
Looking at this I have no idea where this would originate without hunting through the methods being called. I don't think `check_log_delay` should know what this means; whatever method may have caused it should catch it and rethrow as an exception with a meaningful error message (which pod? what query was made? what was the response that was invalid?).